akka-docs/rst/java/stream/stream-quickstart.rst

```diff
@@ -1,7 +1,140 @@
+those numbers out we have to run it:
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#run-source
+
+This line will complement the source with a consumer function—in this example
+we simply print out the numbers to the console—and pass this little stream
+setup to an Actor that runs it. This activation is signaled by having “run” be
+part of the method name; there are other methods that run Akka Streams, and
+they all follow this pattern.
+
+You may wonder where the Actor gets created that runs the stream, and you are
+probably also asking yourself what this ``materializer`` means. In order to get
+this value we first need to create an Actor system:
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#create-materializer
+
+There are other ways to create a materializer, e.g. from an
+:class:`ActorContext` when using streams from within Actors. The
+:class:`Materializer` is a factory for stream execution engines, it is the
+thing that makes streams run—you don’t need to worry about any of the details
+just now apart from that you need one for calling any of the ``run`` methods on
+a :class:`Source`.
+
+The nice thing about Akka Streams is that the :class:`Source` is just a
+description of what you want to run, and like an architect’s blueprint it can
+be reused, incorporated into a larger design. We may choose to transform the
+source of integers and write it to a file instead:
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#transform-source
+
+First we use the ``scan`` combinator to run a computation over the whole
+stream: starting with the number 1 (``BigInteger.ONE``) we multiple by each of
+the incoming numbers, one after the other. This yields the series of factorial
+numbers which we stash away as a :class:`Source` for later reuse—it is
+important to keep in mind that nothing is actually computed yet, this is just a
+description of what we want to have computed once we run the stream. Then we
+convert the resulting series of numbers into a stream of :class:`ByteString`
+objects describing lines in a text file. This stream is then run by attaching a
+file as the receiver of the data. In the terminology of Akka Streams this is
+called a :class:`Sink`.
+
+Reusable Pieces
+---------------
+
+One of the nice parts of Akka Streams—and something that other stream libraries
+do not offer—is that not only sources can be reused like blueprints, all other
+elements can be as well. We can take the file-writing :class:`Sink`, prepend
+the processing steps necessary to get the :class:`ByteString` elements from
+incoming strings and package that up as a reusable piece as well. Since the
+language for writing these streams always flows from left to right (just like
+plain English), we need a starting point that is like a source but with an
+“open” input. In Akka Streams this is called a :class:`Flow`:
+
+.. includecode:: ../code/docs/stream/QuickStartDocTest.java#transform-sink

```

Comment:
one thing that is not nice about this particular example is that due to it taking a filename as parameter, it is not actually reused here (of course it could be, but for newbies this might be a bit confusing).